tests/unit/retrievers/test_cypher.py

```diff
@@ -0,0 +1,299 @@
+#  Copyright (c) "Neo4j"
+#  Neo4j Sweden AB [https://neo4j.com]
+#  #
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  #
+#      https://www.apache.org/licenses/LICENSE-2.0
+#  #
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import unittest
+from unittest.mock import MagicMock, patch
+
+import pytest
+import neo4j
+from neo4j import Record
+
+from neo4j_graphrag.exceptions import (
+    RetrieverInitializationError,
+    SearchValidationError,
+)
+from neo4j_graphrag.retrievers.cypher import CypherRetriever
+from neo4j_graphrag.types import RetrieverResultItem
+
+
+class TestCypherRetriever(unittest.TestCase):
```

Comment:
A bit nitpicking here but we do not use class-based tests in this code base, only functions. So unless there is a strong reason to use a class, I'd prefer we stick to using functions :)